### PR TITLE
Hide FileSharing API from stable build

### PR DESCRIPTION
**Pull request by prprabhu-ms**
# What

Cherry pick of #1744 

# Why

Remove leaked beta-only API.

### CODE REVIEW

**Review by JamesBurnside — APPROVED** [review]

**Comment by prprabhu-ms** [comment]
Will merge this once CI blesses it, even as I wait on `main` merge. I need this in to start API review discussion again.

**Pull request merged**
